File: src/java/PerfectNumber.java
package java;

public class PerfectNumber {

    public static boolean isPerfectNumber(long num) {
        long sum = 0;

        for (long i = 1; i * i <= num ; i++) {
            if (num % i == 0) {
                sum += i;
                if (i != num/i)
                    sum += num/i;
            }
        }

        return sum-num == num;
    }

}

File: src/C++/PerfectNumber.cpp
#include <iostream>

using namespace std;
bool isPerfectNumber(long n);

bool isPerfectNumber(long n) {
    int sum = 0;
    for (int i = 1; i < n; i++) {
        if (n % i == 0) {
            sum += i;
        }
    }

    return sum == n;
}